Runtime/Scripts/DataCollection/RGDataCollection.cs

```diff
@@ -32,20 +37,56 @@ namespace RegressionGames.DataCollection
      *
      * Once all clients have been disconnected, this class can be reset and all data left over (i.e. screenshots) can
      * be deleted from the system.
-     * 
+     *
      */
     public class RGDataCollection
     {
-        
+        private static readonly ProfilerMarker LogCollectionMarker = new(ProfilerCategory.Ai, "RegressionGames.DataCollection.LogCollector");
+        private static readonly ProfilerMarker SaveTickDataMarker = new(ProfilerCategory.Ai, "RegressionGames.DataCollection.SaveTickData");
+
         private readonly string _sessionName = Guid.NewGuid().ToString();
-        private readonly Dictionary<long, RGBot> _clientIdToBots = new();
-        private readonly Dictionary<long, DateTime> _clientIdStartTimes = new();
-        private readonly ConcurrentDictionary<long, List<Task>> _clientIdToReplayDataTasks = new();
-        private readonly ConcurrentDictionary<long, Task> _clientIdToValidationDataTask = new();
+        private readonly ConcurrentDictionary<long, BotInstanceDataCollectionState> _clientIdToState = new();
         private readonly string _rootPath = Application.persistentDataPath;
         private readonly ConcurrentQueue<long> _screenshotTicksRequested = new();
         private readonly ConcurrentDictionary<long, long> _screenshotTicksCaptured = new();
-        private readonly ConcurrentDictionary<long, RGValidationSummary> _clientIdToValidationSummaries = new();
+
+        private class BotInstanceDataCollectionState
+        {
+            public long clientId;
+            public RGBot bot;
+            public ConcurrentQueue<LogDataPoint> logs = new();
+            public DateTime startTime;
+            public List<Task> replayDataTasks = new();
+            public Task validationDataTask;
+            public Task logFlushTask;
+            public RGValidationSummary validationSummary = new(0,0,0);
+
+            public BotInstanceDataCollectionState(RGBot bot, DateTime startTime)
+            {
+                this.bot = bot;
+                this.startTime = startTime;
+            }
+
+            public void StartCapturingLogs()
+            {
+                Application.logMessageReceivedThreaded += LogMessageReceived;
+            }
+
+            public void StopCapturingLogs()
+            {
+                Application.logMessageReceivedThreaded -= LogMessageReceived;
+            }
+
+            private void LogMessageReceived(string message, string stacktrace, LogType type)
+            {
+                Profiler.BeginSample("RegressionGames.DataCollection.LogCollector");
```

Comment:
This sampling needs to be removed, correct?

Reply:
It doesn't have to be. It's compiled out of non-development builds and it can be useful to allow users to see the impact we have